### ISSUE TITLE
Telescope show & recover hidden terminal error

### ISSUE THREAD

**Issue by GreatestOfAllTimee** [NvChad/NvChad#920]
**Describe the bug**
Show & recover hidden terminal by using `<leader>+W`  end up show an _error message_ before recovery terminal show up.

**To Reproduce**
1. `<leader+h` : open a new horizontal terminal.
2. do something...
3. `jk` and `<leader>+x` : normal mode and hide current terminal.

**Expected behavior**
No error should display.

**Desktop (please complete the following information):**
 - Ubuntu 20.04 LTS
 - Terminal Wezterm & Kitty
 - NVIM v0.6.1 Release

**Additional context**
I believe this problems happened after commit  https://github.com/NvChad/NvChad/commit/1face444b83c89fc5a411e07462bd359d4db4e2b, https://github.com/NvChad/NvChad/commit/e328b86336b9fd9cb2c09e10e298392f1705b257 **or** https://github.com/NvChad/extensions/commit/e59fb35af3345735d3634be661be79284f083d7b. `PackerSync` & `PackerUpdate` do not solve this issue.

**Comment by GreatestOfAllTimee** [comment]
**Gif:**

![Terminal_ErrorMessage](https://user-images.githubusercontent.com/73665622/161036035-8a052328-7bc2-41c5-ae5d-b7e71f6298eb.gif)

**Comment by siduck** [comment]
do packersync

**Comment by GreatestOfAllTimee** [comment]
yes, there is a difference error between first (gif) and the second (image).

**Second Image Error Traceback:**

```
E5108: Error executing lua ...ite/pack/packer/start/extensions/lua/nvchad/terminal.lua:13: attempt to perform arithmetic on a nil value
stack traceback:

...ite/pack/packer/start/extensions/lua/nvchad/terminal.lua:13: in function 'get_dims'
...ite/pack/packer/start/extensions/lua/nvchad/terminal.lua:16: in function 'get_cmds'
...ite/pack/packer/start/extensions/lua/nvchad/terminal.lua:54: in function 'new_or_toggle'
...ker/start/extensions/lua/telescope/_extensions/terms.lua:102: in function 'run_replace_or_original'
...k/packer/opt/telescope.nvim/lua/telescope/actions/mt.lua:65: in function 'key_func'
...ack/packer/opt/telescope.nvim/lua/telescope/mappings.lua:242: in function 'execute_keymap'
```

**Comment by siduck** [comment]
@GreatestOfAllTimee  so leader shift w, produces this error right?

**Comment by GreatestOfAllTimee** [comment]
Yes it is still produces an error.

**Comment by siduck** [comment]
https://user-images.githubusercontent.com/59060246/161206244-e4eb90a5-a5bb-4887-8a54-7660e041ff95.mp4

your config's outdated probably or you forgot to do a packersync

**Comment by GreatestOfAllTimee** [comment]
what's you mean by outdated? I'm in a latest Nvchad commit (which are https://github.com/NvChad/NvChad/commit/63d54926a4a80e289d4c49cda70ab7b18cfa64d2) and my `custom/chadrc.lua` same as [`siduck/dotfiles/chadrc.lua`](https://github.com/siduck/dotfiles/blob/master/nvchad/custom/chadrc.lua).

### **Issue:**

![Terminal_ErrorMessage2](https://user-images.githubusercontent.com/73665622/161212185-06eb58e6-a10e-45dd-a7c9-9b837e83d9fa.gif)

**Comment by siduck** [comment]
oh i get it now, 

leader + h -> no error 
leader  + H -> will give error 

https://user-images.githubusercontent.com/59060246/161216663-31c64b76-2daf-4f88-afcf-919dbef57181.mp4

Idk whats causing this issue, maybe @zbirenbaum knows

**Comment by GreatestOfAllTimee** [comment]
Yah, there is multi bugs on this issue. Recently i found a new issue: `leader + v` (open new vertical terminal) and hide via `leader + x`, if you create a new vertical terminal `leader + v` _again_, an error messages will popup.

This issue seems like critical on Nvchad, hope @zbirenbaum or someone fix this before release a new feature. Neovim 0.7 is comming soon btw, so im excited to see whats coming on Nvchad next. No pressure 👌

**Comment by zbirenbaum** [comment]
For the part about H/V vs h/v, if it bothers you that much and you can't stop pressing shift, just map leader H/V to be the same as leader h/v. I'm not even sure how this is possible, but my guess is it's an upstream thing. It's also going to be way easier to fix stuff like this once I have liberal use of vim.keymap when .7 hits.

> Yah, there is multi bugs on this issue. Recently i found a new issue: `leader + v` (open new vertical terminal) and hide via `leader + x`, if you create a new vertical terminal `leader + v` _again_, an error messages will popup.
> 
> This issue seems like critical on Nvchad, hope @zbirenbaum or someone fix this before release a new feature. Neovim 0.7 is comming soon btw, so im excited to see whats coming on Nvchad next. No pressure ok_hand

I just pushed a fix to this. I was already verifying that the buffer still existed and wasn't closed in an unexpected way, but now I do the same for windows too. No longer errors.

**Comment by GreatestOfAllTimee** [comment]
Thanks for your quick response, I can confirm that `<leader> + v` (open new vertical terminal) fixed for now. But I think the main issue is https://github.com/NvChad/NvChad/issues/920#issuecomment-1084396397 which are not work as we expected.

**Comment by siduck** [comment]
@zbirenbaum we were able to unhide terminals spawned by telescope with the old implementation but now we're not able to unhide termiinals spawned by leader + H or leader + V , but we can unhide terminals spawned by leader + h or leader + v

![image](https://user-images.githubusercontent.com/59060246/162611156-fe90ef70-565a-4f2f-8740-3b07465ea7ac.png)
I get this error when unhiding terminals that were spawned by leader + H or leader +V . This isnt an upstream issue ig! 

Maybe the nvchad/terminal.lua is only made to    manage terminals (i.e unhide them) spawned by leader + h /v

**Comment by siduck** [comment]
I see that going back to this commit https://github.com/NvChad/extensions/blob/c4ee918b8f6e6183de9696f0a3db06945f56eea8/lua/telescope/_extensions/terms.lua 

fixes the issue

**Comment by GreatestOfAllTimee** [comment]
![Terminal_ErrorMessage3](https://user-images.githubusercontent.com/73665622/162611197-c5343d06-ac80-42b6-87ab-8ad0df911322.gif)


1) Open a new horizontal split terminal: `<leader> + h`
2) `j + k` back to normal mode.
3) Hide terminal `<leader> + x`.
4) Show & recover hidden terminal buffers in a telescope picker: `<leader> + W`
5) an error messages as above popup.

> All the key map I mention above are coming from Nvchad default key.

**Comment by zbirenbaum** [comment]
> @zbirenbaum we were able to unhide terminals spawned by telescope with the old implementation but now we're not able to unhide termiinals spawned by leader + H or leader + V , but we can unhide terminals spawned by leader + h or leader + v
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/59060246/162611156-fe90ef70-565a-4f2f-8740-3b07465ea7ac.png)
> 
> I get this error when unhiding terminals that were spawned by leader + H or leader +V . This isnt an upstream issue ig! 
> 
> 
> 
> Maybe the nvchad/terminal.lua is only made to    manage terminals (i.e unhide them) spawned by leader + h /v 

Let me make sure I have this correct. Telescope works fine as long as leader H and V are never touched. Terminals spawned by leader h and v are fine, and can be hidden with leader H and V. 

It's irrelevant what you have bound to call terminal.lua, what matters is that the very first instance of the terminal is created by calling terminal.lua.

I can do all the error handling in the world to check if the user closes their terminal some way outside of the plugin calls. There's no way for me to get information about the location or type of terminal unless the spawn it through the plugin.

To be perfectly honest I'm not sure why leader H and V do anything at all.

**Comment by zbirenbaum** [comment]
> I see that going back to this commit  
> 
> https://github.com/NvChad/extensions/blob/c4ee918b8f6e6183de9696f0a3db06945f56eea8/lua/telescope/_extensions/terms.lua
> 
> fixes the issue

Why was it changed after that commit? If I changed it to fix something else we shouldn't break that to fix this.

**Comment by siduck** [comment]
@zbirenbaum In a nutshell, this file doesnt has to handle the nvchad_terminal stuff at all since we can easily toggle terminals by their mappings 
![image](https://user-images.githubusercontent.com/59060246/162611527-5c69efc9-5c94-4783-9a74-2e26bc0771ac.png)


But we can spawn infinite number of terminals via leader + H or V so it gets difficult to toggle them by a mapping so we hide them and pick them by telescope. So i'll revert to a lsat working commit as it fixes the current issue and doesnt break any functionality with the new toggle term feature made by u

**Comment by zbirenbaum** [comment]
> @zbirenbaum In a nutshell, this file doesnt has to handle the nvchad_terminal stuff at all since we can easily toggle terminals by their mappings 
> 
> ![image](https://user-images.githubusercontent.com/59060246/162611527-5c69efc9-5c94-4783-9a74-2e26bc0771ac.png)
> 
> 
> 
> 
> 
> But we can spawn infinite number of terminals via leader + H or V so it gets difficult to toggle them by a mapping so we hide them and pick them by telescope. So i'll revert to a lsat working commit as it fixes the current issue and doesnt break any functionality with the new toggle term feature made by u

I don't think we should encourage multiple terminal mappings by default, that will cause confusion. Does this mean that leader H and V are official mappings as a work around for multiple terms?

I had telescope working with the toggle terms, not sure what happened.

**Comment by zbirenbaum** [comment]
> @zbirenbaum In a nutshell, this file doesnt has to handle the nvchad_terminal stuff at all since we can easily toggle terminals by their mappings 
> 
> ![image](https://user-images.githubusercontent.com/59060246/162611527-5c69efc9-5c94-4783-9a74-2e26bc0771ac.png)
> 
> 
> 
> 
> 
> But we can spawn infinite number of terminals via leader + H or V so it gets difficult to toggle them by a mapping so we hide them and pick them by telescope. So i'll revert to a lsat working commit as it fixes the current issue and doesnt break any functionality with the new toggle term feature made by u

If you want to have multiple terminals I think I already made it so that you can call new() directly. Please don't revert to that and have ways to open terminals outside of the terminal plugin. I think that's what's causing all the confusion here.

**Comment by siduck** [comment]
> > @zbirenbaum In a nutshell, this file doesnt has to handle the nvchad_terminal stuff at all since we can easily toggle terminals by their mappings
> > ![image](https://user-images.githubusercontent.com/59060246/162611527-5c69efc9-5c94-4783-9a74-2e26bc0771ac.png)
> > But we can spawn infinite number of terminals via leader + H or V so it gets difficult to toggle them by a mapping so we hide them and pick them by telescope. So i'll revert to a lsat working commit as it fixes the current issue and doesnt break any functionality with the new toggle term feature made by u
> 
> I don't think we should encourage multiple terminal mappings by default, that will cause confusion. Does this mean that leader H and V are official mappings as a work around for multiple terms?
> 
> I had telescope working with the toggle terms, not sure what happened.

https://user-images.githubusercontent.com/59060246/162611904-0ffc5937-5629-467d-af16-8cb2ac9e7ec7.mp4

telescope isnt needed with toggle terms as they're can be easily made unhidden by mappings unlike multiple terminals spawned by leader + H/V

**Comment by zbirenbaum** [comment]
> > > @zbirenbaum In a nutshell, this file doesnt has to handle the nvchad_terminal stuff at all since we can easily toggle terminals by their mappings
> 
> > > ![image](https://user-images.githubusercontent.com/59060246/162611527-5c69efc9-5c94-4783-9a74-2e26bc0771ac.png)
> 
> > > But we can spawn infinite number of terminals via leader + H or V so it gets difficult to toggle them by a mapping so we hide them and pick them by telescope. So i'll revert to a lsat working commit as it fixes the current issue and doesnt break any functionality with the new toggle term feature made by u
> 
> > 
> 
> > I don't think we should encourage multiple terminal mappings by default, that will cause confusion. Does this mean that leader H and V are official mappings as a work around for multiple terms?
> 
> > 
> 
> > I had telescope working with the toggle terms, not sure what happened.
> 
> 
> 
> https://user-images.githubusercontent.com/59060246/162611904-0ffc5937-5629-467d-af16-8cb2ac9e7ec7.mp4
> 
> 
> 
> telescope isnt needed with toggle terms as they're can be easily made unhidden by mappings unlike multiple terminals spawned by leader + H/V
> 
> 
> 
>  

@siduck please read the comment directly above this one. I implemented infinite terminal spawn already as part of my terms plugin.

**Comment by siduck** [comment]
> ![Terminal_ErrorMessage3](https://user-images.githubusercontent.com/73665622/162611197-c5343d06-ac80-42b6-87ab-8ad0df911322.gif)
> 
> 1. Open a new horizontal split terminal: `<leader> + h`
> 2. `j + k` back to normal mode.
> 3. Hide terminal `<leader> + x`.
> 4. Show & recover hidden terminal buffers in a telescope picker: `<leader> + W`
> 5. an error messages as above popup.
> 
> > All the key map I mention above are coming from Nvchad default key.

this should be in the case of leader + H/V. but its fixed now 

if i do the same as your instructions, I get this message 
![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)

**Comment by siduck** [comment]
> > > > @zbirenbaum In a nutshell, this file doesnt has to handle the nvchad_terminal stuff at all since we can easily toggle terminals by their mappings
> > 
> > 
> > > > ![image](https://user-images.githubusercontent.com/59060246/162611527-5c69efc9-5c94-4783-9a74-2e26bc0771ac.png)
> > 
> > 
> > > > But we can spawn infinite number of terminals via leader + H or V so it gets difficult to toggle them by a mapping so we hide them and pick them by telescope. So i'll revert to a lsat working commit as it fixes the current issue and doesnt break any functionality with the new toggle term feature made by u
> > 
> > 
> > > 
> > 
> > 
> > > I don't think we should encourage multiple terminal mappings by default, that will cause confusion. Does this mean that leader H and V are official mappings as a work around for multiple terms?
> > 
> > 
> > > 
> > 
> > 
> > > I had telescope working with the toggle terms, not sure what happened.
> > 
> > 
> > 
> >   
> >     
> >     
> > 
> >     simplescreenrecorder-2022-04-10_15.00.50.mp4
> >     
> >   
> > 
> >   
> > 
> >   
> > 
> > 
> > telescope isnt needed with toggle terms as they're can be easily made unhidden by mappings unlike multiple terminals spawned by leader + H/V
> 
> @siduck please read the comment directly above this one. I implemented infinite terminal spawn already as part of my terms plugin.

oh oops i missed that part, i'll try looking onto it

**Comment by siduck** [comment]
soo 
![image](https://user-images.githubusercontent.com/59060246/162612133-d65c2a5c-ea55-4f0a-bbd2-e59514eab3c7.png)
?

**Comment by zbirenbaum** [comment]
![image](https://user-images.githubusercontent.com/32016110/162612071-68ea2110-29bc-462d-893b-0204d637c90c.png)

@siduck you can see here that M.new is returned by terminal and directly callable.

Make a new mapping to call it with the required args instead of new or toggle

**Comment by siduck** [comment]
> soo ![image](https://user-images.githubusercontent.com/59060246/162612133-d65c2a5c-ea55-4f0a-bbd2-e59514eab3c7.png) ?

ok i tried this and it does spawn multiple terminals but im unable to hide them as leader +x just kills them directly and telescope as a result has nothing to pick 

![image](https://user-images.githubusercontent.com/59060246/162612298-17055a67-4ccf-4a6c-8a75-6f7e683e68d0.png)

**Comment by siduck** [comment]
Also there's a difference in the terminal window size 

nvchad_terminal.new() 

![image](https://user-images.githubusercontent.com/59060246/162612333-5537d9f3-5dcd-4c35-90cc-3b5881720a97.png)

our old implementation of spawning multiple terminals 

![image](https://user-images.githubusercontent.com/59060246/162612353-b72ccd55-4ce7-4908-8c41-9619b1caf49b.png)

**Comment by zbirenbaum** [comment]
> > soo ![image](https://user-images.githubusercontent.com/59060246/162612133-d65c2a5c-ea55-4f0a-bbd2-e59514eab3c7.png) ?
> 
> 
> 
> ok i tried this and it does spawn multiple terminals but im unable to hide them as leader +x just kills them directly and telescope as a result has nothing to pick 
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/59060246/162612298-17055a67-4ccf-4a6c-8a75-6f7e683e68d0.png)
> 
> 

If leader x is doing more than closing terminals then wait until .7 drops. Once it does I can make leader-x call a function which behaves differently depending on if it's a terminal or not. What you want is a pain to implement with .6 mappings.

The reason the sizes are different is because your terminal mapping spawns fixed size, while mine is based on a ratio of the current window's size. My way is debatably a better method in any case other than this particular one.

I can make an override parameter you can pass to M.new to get a fixed size spawn but not until tomorrow at the earliest.

**Comment by siduck** [comment]
alright, lets wait for .7. Until then there's a temp revert fix for now @GreatestOfAllTimee

**Comment by zbirenbaum** [comment]
> Also there's a difference in the terminal window size 
> 
> 
> 
> nvchad_terminal.new() 
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/59060246/162612333-5537d9f3-5dcd-4c35-90cc-3b5881720a97.png)
> 
> 
> 
> our old implementation of spawning multiple terminals 
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/59060246/162612353-b72ccd55-4ce7-4908-8c41-9619b1caf49b.png)
> 
> 

By the way, I think you could also achieve this even with the current mappings by opening a horizontal toggle term, opening a vertical toggle term from inside the horizontal, and then opening a new horizontal from within each of the two spawned terminals.

**Comment by GreatestOfAllTimee** [comment]
> if i do the same as your instructions, I get this message ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)

https://github.com/NvChad/extensions/commit/84dac985d8d4ff4c79c0cf439cc5a06201282e6c latest commit I think fixed some show and hide terminal issue. But weird when using `<leader> + v` and unhide _no terminal buffer are open_:

![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)

**Comment by zbirenbaum** [comment]
> > if i do the same as your instructions, I get this message ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)
> 
> 
> 
> https://github.com/NvChad/extensions/commit/84dac985d8d4ff4c79c0cf439cc5a06201282e6c latest commit I think fixed some show and hide terminal issue. But weird when using `<leader> + v` and unhide _no terminal buffer are open_:
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)

Yeah, you can't have it both ways for right now.

**Comment by siduck** [comment]
> > if i do the same as your instructions, I get this message ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)
> 
> [NvChad/extensions@84dac98](https://github.com/NvChad/extensions/commit/84dac985d8d4ff4c79c0cf439cc5a06201282e6c) latest commit I think fixed some show and hide terminal issue. But weird when using `<leader> + v` and unhide _no terminal buffer are open_:
> 
> ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)

@GreatestOfAllTimee leader + x hide's terminals opened by Ctrl + v or h ( mappings have been changed in recent commit ) and these mappings are used for spawning multiple terminals 


Where as terminals opened by leader + h or v can be get hidden or toggled by their mappings itself so leader +x wont work on them (  and telescope  picker doesnt work on it )

**Comment by zbirenbaum** [comment]
> > > if i do the same as your instructions, I get this message ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)
> 
> > 
> 
> > [NvChad/extensions@84dac98](https://github.com/NvChad/extensions/commit/84dac985d8d4ff4c79c0cf439cc5a06201282e6c) latest commit I think fixed some show and hide terminal issue. But weird when using `<leader> + v` and unhide _no terminal buffer are open_:
> 
> > 
> 
> > ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)
> 
> 
> 
> @GreatestOfAllTimee leader + x hide's terminals opened by Ctrl + v or h ( mappings have been changed in recent commit ) and these mappings are used for spawning multiple terminals 
> 
> 
> 
> 
> 
> Where as terminals opened by leader + h or v can be get hidden or toggled by their mappings itself so leader +x wont work on them (  and telescope  picker doesnt work on it ) 
> 
> 

ctrl+v is visual block mode and ctrl+h is often used for moving between buffers or to the beginning of line. Please don't remap those. Ctrl+v especially, that's super important for multi line append/inserts.

Ctrl+s and ctrl+x are much better options.

**Comment by siduck** [comment]
> > > > if i do the same as your instructions, I get this message ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)
> > 
> > 
> > > 
> > 
> > 
> > > [NvChad/extensions@84dac98](https://github.com/NvChad/extensions/commit/84dac985d8d4ff4c79c0cf439cc5a06201282e6c) latest commit I think fixed some show and hide terminal issue. But weird when using `<leader> + v` and unhide _no terminal buffer are open_:
> > 
> > 
> > > 
> > 
> > 
> > > ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)
> > 
> > 
> > @GreatestOfAllTimee leader + x hide's terminals opened by Ctrl + v or h ( mappings have been changed in recent commit ) and these mappings are used for spawning multiple terminals
> > Where as terminals opened by leader + h or v can be get hidden or toggled by their mappings itself so leader +x wont work on them (  and telescope  picker doesnt work on it )
> 
> ctrl+v is visual block mode and ctrl+h is often used for moving between buffers or to the beginning of line. Please don't remap those. Ctrl+v especially, that's super important for multi line append/inserts.
> 
> Ctrl+s and ctrl+x are much better options.
> 
> Even if it's a bit uncomfortable for the vertical one, i would even prefer alt h and alt v to those mappings until we get this sorted out and unified around .7 release.

whoa didnt know :cold_sweat: , reverted them back

**Comment by zbirenbaum** [comment]
> > > > > if i do the same as your instructions, I get this message ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > [NvChad/extensions@84dac98](https://github.com/NvChad/extensions/commit/84dac985d8d4ff4c79c0cf439cc5a06201282e6c) latest commit I think fixed some show and hide terminal issue. But weird when using `<leader> + v` and unhide _no terminal buffer are open_:
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > ![image](https://user-images.githubusercontent.com/59060246/162612028-7c55e74d-ed5a-49ff-965a-a783beb1bf2a.png)
> 
> > > 
> 
> > > 
> 
> > > @GreatestOfAllTimee leader + x hide's terminals opened by Ctrl + v or h ( mappings have been changed in recent commit ) and these mappings are used for spawning multiple terminals
> 
> > > Where as terminals opened by leader + h or v can be get hidden or toggled by their mappings itself so leader +x wont work on them (  and telescope  picker doesnt work on it )
> 
> > 
> 
> > ctrl+v is visual block mode and ctrl+h is often used for moving between buffers or to the beginning of line. Please don't remap those. Ctrl+v especially, that's super important for multi line append/inserts.
> 
> > 
> 
> > Ctrl+s and ctrl+x are much better options.
> 
> > 
> 
> > Even if it's a bit uncomfortable for the vertical one, i would even prefer alt h and alt v to those mappings until we get this sorted out and unified around .7 release.
> 
> 
> 
> whoa didnt know :cold_sweat: , reverted them back

All good, thanks for swapping it back 👍

If you haven't tried it before, press ctrl v, use h and h to select multiple lines, then press shift i to insert before them or shift a to append after them. 

It will look like you are only typing on one line, but press jk or escape and it will apply whatever you typed to all of them. Super useful at times for refactoring.

**Comment by siduck** [comment]
@GreatestOfAllTimee this is fixed now, ik its late!

**Comment by GreatestOfAllTimee** [comment]
great to know that, I'm waiting for [v2.0 (GigaChad)](https://github.com/NvChad/NvChad/milestone/2)  to be released before I update Nvchad or not even do `:PackerSync` is because I'm aware of this issue https://github.com/NvChad/NvChad/issues/996. Take your time 👍

**Comment by siduck** [comment]
> great to know that, I'm waiting for [v2.0 (GigaChad)](https://github.com/NvChad/NvChad/milestone/2) to be released before I update Nvchad or not even do `:PackerSync` is because I'm aware of this issue #996. Take your time +1

no thats not an issue, I was just showing a way to reduce startuptime, that change is not on repo so update your nvchad now itself!
